[NEMO-76] Rename TaskGroup to Task

johnyangk opened the pull request:
JIRA: [NEMO-76: Rename TaskGroup to Task](https://issues.apache.org/jira/projects/NEMO/issues/NEMO-76)

**Major changes:**
- Appropriately renames things like 'TaskGroup', 'taskGroup', 'Task group', and 'task group'

**Minor changes to note:**
- Removes the legacy TaskState

**Tests for the changes:**
- This PR only changes names, and removes TaskState that isn't used anywhere
- The existing tests continue to pass

**Other comments:**
- The legacy 'Task' will be handled with https://issues.apache.org/jira/browse/NEMO-79
- Once this PR is merged, I'll work on that ASAP to remove the confusion between the two 'Task's.

resolves [NEMO-76](https://issues.apache.org/jira/projects/NEMO/issues/NEMO-76)

jeongyooneo commented on runtime/master/src/main/java/edu/snu/nemo/runtime/master/JobStateManager.java:
```diff
@@ -131,36 +131,36 @@ private void initializeComputationStates() {
-        taskGroupInternalDag.getVertices().forEach(task -> {
-          final List<RuntimeEdge<Task>> internalOutgoingEdges = taskGroupInternalDag.getOutgoingEdgesOf(task);
+      taskIdsForStage.forEach(taskId -> {
+        final DAG<Task, RuntimeEdge<Task>> taskInternalDag = physicalStage.getTaskDag();
```
taskInternalDAG -> taskDAG?

jeongyooneo commented on runtime/master/src/main/java/edu/snu/nemo/runtime/master/JobStateManager.java:
```diff
@@ -260,39 +260,38 @@ public synchronized void onStageStateChanged(final String stageId, final StageSt
    * and the call to this method is initiated in {@link edu.snu.nemo.runtime.master.scheduler.BatchSingleJobScheduler}
    * when the message/event is received.
-   * A task group completion implies completion of all its tasks.
+   * A task completion implies completion of all its tasks.
```
Maybe remove this line?

jeongyooneo commented on runtime/master/src/main/java/edu/snu/nemo/runtime/master/JobStateManager.java:
```diff
@@ -260,39 +260,38 @@ public synchronized void onStageStateChanged(final String stageId, final StageSt
+  public synchronized void onTaskStateChanged(final String taskId, final TaskState.State newState) {
+    final StateMachine taskState = idToTaskStates.get(taskId).getStateMachine();
+    final String stageId = RuntimeIdGenerator.getStageIdFromTaskId(taskId);
     LOG.debug("Task Group State Transition: id {}, from {} to {}",
```
Task Group -> Task?

jeongyooneo commented on runtime/master/src/test/java/edu.snu.nemo.runtime.master/scheduler/SourceLocationAwareSchedulingPolicyTest.java:
```diff
@@ -46,15 +46,15 @@ private static ExecutorRepresenter mockExecutorRepresenter(final String executor
 
     // Prepare test scenario
-    final ScheduledTaskGroup tg = CreateScheduledTaskGroup.withReadablesWithSourceLocations(
+    final ScheduledTask tg = CreateScheduledTask.withReadablesWithSourceLocations(
```
tg -> t?

jeongyooneo commented on runtime/master/src/test/java/edu.snu.nemo.runtime.master/scheduler/SourceLocationAwareSchedulingPolicyTest.java:
```diff
@@ -70,43 +70,43 @@ public void testSourceLocationAwareSchedulingNotAvailable() {
     final SchedulingPolicy schedulingPolicy = new SourceLocationAwareSchedulingPolicy();
     // Prepare test scenario
-    final ScheduledTaskGroup tg0 = CreateScheduledTaskGroup.withReadablesWithSourceLocations(
+    final ScheduledTask tg0 = CreateScheduledTask.withReadablesWithSourceLocations(
```
tg0 -> t0 and to all other local variables starting with tg?

jeongyooneo commented on runtime/master/src/test/java/edu.snu.nemo.runtime.master/scheduler/SourceLocationAwareSchedulingPolicyTest.java:
```diff
@@ -70,43 +70,43 @@ public void testSourceLocationAwareSchedulingNotAvailable() {
-  private static final class CreateScheduledTaskGroup {
-    private static final AtomicInteger taskGroupIndex = new AtomicInteger(0);
+  private static final class CreateScheduledTask {
+    private static final AtomicInteger tgIndex = new AtomicInteger(0);
```
tIndex?

jeongyooneo commented on runtime/master/src/test/java/edu.snu.nemo.runtime.master/scheduler/SourceLocationAwareSchedulingPolicyTest.java:
```diff
@@ -70,43 +70,43 @@ public void testSourceLocationAwareSchedulingNotAvailable() {
       readables.forEach(readable -> readableMap.put(String.format("TASK-%d", taskIndex.getAndIncrement()),
           readable));
-      when(mockInstance.getTaskGroupId()).thenReturn(String.format("TG-%d", taskGroupIndex.getAndIncrement()));
+      when(mockInstance.getTaskId()).thenReturn(String.format("TG-%d", tgIndex.getAndIncrement()));
```
"T-%d"?

jeongyooneo commented on tests/src/test/java/edu/snu/nemo/tests/runtime/executor/TaskExecutorTest.java:
```diff
@@ -88,11 +88,11 @@ public void setUp() throws Exception {
   @Test(timeout=2000)
   public void testSourceTask() throws Exception {
-    // Create a task group only having a source task.
+    // Create a task only having a source task.
```
Maybe remove this comment if not necessary?

jeongyooneo commented on tests/src/test/java/edu/snu/nemo/tests/runtime/executor/TaskExecutorTest.java:
```diff
@@ -145,7 +145,7 @@ public Iterable read() throws Exception {
   @Test//(timeout=2000)
   public void testOperatorTask() throws Exception {
-    // Create a task group with two operator tasks.
+    // Create a task with two operator tasks.
```
Maybe remove this comment if not necessary?

johnyangk commented:
@jeongyooneo I've addressed your comments except the ones for TaskDataHandler and 'pipelined' tasks, which I'd like to address while working on `NEMO-79`.

jeongyooneo commented:
@johnyangk Of course!
